src/crimson/os/seastore/seastore.cc

```diff
@@ -1715,15 +1901,14 @@ seastar::future<> SeaStore::write_meta(const std::string& key,
               return transaction_manager->submit_transaction(t);
             });
           });
-	}).safe_then([this, &key, &value] {
-	  return mdstore->write_meta(key, value);
 	});
       }).handle_error(
 	crimson::ct_error::assert_all{"Invalid error in SeaStore::write_meta"}
       );
 }
 
-seastar::future<std::tuple<int, std::string>> SeaStore::read_meta(const std::string& key)
+seastar::future<std::tuple<int, std::string>>
+SeaStore::read_meta(const std::string& key)
 {
   LOG_PREFIX(SeaStore::read_meta);
```

Comment:
ceph_assert(seastar::this_shard_id() == PRIMARY_CORE);

Please also scan the code to add this assert to other public interfaces that should only be available in PRIMARY_CORE.